src/mat/impls/inv/matinv.c

```diff
@@ -50,14 +50,14 @@ static PetscErrorCode MatInvGetRegularizationType_Inv(Mat imat,MatRegularization
   PetscFunctionReturn(0);
 }
 
+//TODO MOVE THIS TO permonmatnullspace.c
+//TODO Passing the PC is awkward. We will want to avoid that but it needs deeper refactoring.
```

Comment:
We could have MatComputeNullSpaceMat() interface function that calls implementation-specific functions and which defaults to MatComputeNullSpaceMat_Default(Mat,PC,...). MatInv would then have only a wrapper implementation to pass PC to _Default().

Reply:
I'm not keen to do that. My ultimate goal is rather to get really rid of the PC argument carefully, including that [terrible play of options database parameters](https://github.com/permon/permon/pull/47/files/a9caa50217df0933e5cd9f3957338052edb5db57#diff-99556492510148a43536b8fdd4e88e24R100). It is misconception posed by current state of affairs in PETSc.

I think keeping this is a reasonable temporary solution and I will create an issue.

Reply:
> including that [terrible play of options database parameters](https://github.com/permon/permon/pull/47/files/a9caa50217df0933e5cd9f3957338052edb5db57#diff-99556492510148a43536b8fdd4e88e24R100). It is misconception posed by current state of affairs in PETSc.
> 
What are you reffering to?

Reply:
Sorry, probably a wrong link. I meant the use of PetscOptionsInsertString in https://github.com/permon/permon/blob/a9caa50217df0933e5cd9f3957338052edb5db57/src/mat/impls/inv/matinv.c#L103

Reply:
I am not sure how do you want to remove either PC or the OptionsInsert.

Reply:
Again, this is caused by the current PETSc design which is obviously limiting. We would need a way to pass settings to the MUMPS factor before the actual factorization which is a legitimate request. I can think of several ways of doing this but please let's not solve it here.